src/plugins/data/common/search/aggs/buckets/shard_delay.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BucketAggType } from './bucket_agg_type';
+import { BaseAggParams } from '../types';
+
+export const SHARD_DELAY_AGG_NAME = 'shard_delay';
+
+export interface AggParamsShardDelay extends BaseAggParams {
+  delay: number;
```

Comment:
nit: So far in most other places we've set properties of agg param interfaces as optional if there's a default that is applied within the agg type.

This is a little confusing because these types are shared with the expression functions, which typically don't have the idea of optional args since default values are injected by the interpreter. But in general I would suggest keeping this in line with the agg type itself:
```suggestion
  delay?: number;
```

In practice it probably doesn't matter a ton for this particular agg since it is only here for testing purposes, but wanted to mention it.